File: system/multitier/backend-typescript/src/api/controller/admin.controller.ts
import { Controller, HttpCode, Param, Post } from '@nestjs/common';
import { OrderService } from '../../core/services/order.service';

@Controller('api/admin')
export class AdminController {
  constructor(private readonly orderService: OrderService) {}

  @Post('recall/:sku')
  @HttpCode(200)
  async recallSku(@Param('sku') sku: string) {
    return this.orderService.recallSku(sku);
  }
}

File: system/multitier/backend-java/src/main/java/com/mycompany/myshop/backend/api/controller/AdminController.java
package com.mycompany.myshop.backend.api.controller;

import com.mycompany.myshop.backend.core.dtos.RecallSkuResponse;
import com.mycompany.myshop.backend.core.services.OrderService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/admin")
public class AdminController {

    private final OrderService orderService;

    public AdminController(OrderService orderService) {
        this.orderService = orderService;
    }

    @PostMapping("/recall/{sku}")
    public ResponseEntity<RecallSkuResponse> recallSku(@PathVariable String sku) {
        var response = orderService.recallSku(sku);
        return ResponseEntity.ok(response);
    }
}
